### ISSUE TITLE
Modelmaker: input/output shapes gets altered when converting from .pb to .tflite

### ISSUE THREAD

**Issue by ra9hur** [tensorflow/tensorflow#57931]
### 1. System information

- Colab environment
- tflite-model-maker-nightly
- tflite-support

### 2. Issue
- Train a model using,
model = object_detector.create()

- Export to tflite using FP16 quantization
config = QuantizationConfig.for_float16()
model.export(export_dir='./', tflite_filename='check_fp16.tflite', quantization_config=config)

- Check for input shape, input type, out shape, output type
Input Shape: [  1 320 320   3]
Input Type: <class 'numpy.float32'>
Output Shape: []
Output Type: <class 'numpy.float32'>

Next, create a tflite file through .pb file and then check for input / output shapes.

- Export to .pb format
model.export(export_dir='.', saved_model_filename='android_wts', export_format=ExportFormat.SAVED_MODEL)
OR
model._export_saved_model('android_wts_saved')

- Convert .pb format to tflite using fp16 quantization
file_path = './android_wts'
tf_lite_converter = tf.lite.TFLiteConverter.from_saved_model(file_path, signature_keys=['serving_default'])

tf_lite_converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS, tf.lite.OpsSet.SELECT_TF_OPS]

tf_lite_converter.optimizations = [tf.lite.Optimize.DEFAULT]   # Have tried enabling this parameter as well
tf_lite_converter.target_spec.supported_types = [tf.float16]
tflite_model = tf_lite_converter.convert()

TF_LITE_MODEL_FILE_NAME = "/currdrive/road_hero/model_noquant.tflite"

with tf.io.gfile.GFile(TF_LITE_MODEL_FILE_NAME, 'wb') as f:
  f.write(tflite_model)

- Check for input shape, input type, out shape, output type
Input Shape: [1 1 1 3]
Input Type: <class 'numpy.uint8'>
Output Shape: [1 1]
Output Type: <class 'numpy.float32'>

To summarize, there is a mismatch in [input shape, input type, out shape, output type]. Suspect, some information is lost, while creating a file with .pb format.

Can you please look into this.

### 3. Reference colab notebooks

1)  Access to Colab notebook
[Colab Notebook Link](https://colab.research.google.com/drive/1QlXjo3aXsOVBbr6uTTbHC2qECceMO_5V#scrollTo=pSidbHq4Pt6w)

Have given required access to colab notebook.

2) Attaching the colab file for your reference.
[modelmaker_quantize.zip](https://github.com/tensorflow/tensorflow/files/9690409/modelmaker_quantize.zip)

**Comment by ra9hur** [comment]
@sachinprasad, - Any update on this issue ?

**Comment by ra9hur** [comment]
@miaout17 - any update please ?

**Comment by ra9hur** [comment]
@miaout17, @sachinprasad, @mohantym  - It has been more than 3 weeks. I have raised 3 issues in the last week of Sep and there has been no response for any of these. Can someone please respond ?

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.
